styx-scheduler-service/src/main/java/com/spotify/styx/api/SchedulerResource.java

```diff
@@ -154,19 +168,48 @@ public SchedulerResource(
     return Response.forPayload(workflow);
   }
 
+  private Response<WorkflowInstance> haltWorkflowInstance(WorkflowInstance workflowInstance) {
+    final Event event = Event.halt(workflowInstance);
+    return Response.forStatus(eventInjectorHelper(event)).withPayload(workflowInstance);
+  }
+
+  private Response<WorkflowInstance> retryWorkflowInstanceAfter(RequestContext rc,
+                                                                WorkflowInstance workflowInstance) {
+    final long delay;
+    try {
+      delay = Long.parseLong(rc.request().parameter("delay").orElse("0"));
+    } catch (NumberFormatException e) {
+      return Response.forStatus(BAD_REQUEST.withReasonPhrase(
+          "Delay parameter could not be parsed"));
+    }
+    final Event event = Event.retryAfter(workflowInstance, delay);
+    return Response.forStatus(eventInjectorHelper(event)).withPayload(workflowInstance);
+  }
 
   private Response<Event> injectEvent(Event event) {
-    if (!stateManager.isActiveWorkflowInstance(event.workflowInstance())) {
-      return Response.forStatus(BAD_REQUEST.withReasonPhrase("Workflow instance not found"));
+    if ("dequeue".equals(EventUtil.name(event))) {
+      // For backwards compatibility
+      return Response.forStatus(eventInjectorHelper(
+          Event.retryAfter(event.workflowInstance(), 0L))).withPayload(event);
+    } else if ("halt".equals(EventUtil.name(event))) {
+      // For backwards compatibility
+      return Response.forStatus(eventInjectorHelper(event));
+    } else {
+      return Response.forStatus(NOT_ACCEPTABLE.withReasonPhrase(
```

Comment:
`406 Not Acceptable` seems to have to do with content type negotiation and accept headers, which I guess is not applicable here.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.7

Use `400 Bad Request` instead ?

Reply:
Good point, changed it to 400